cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -158,20 +152,6 @@ func removeOldDNSDeploymentIfAnotherDNSIsUsed(cfg *kubeadmapi.InitConfiguration,
 	}, 10)
 }
 
-func upgradeToSelfHosting(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, dryRun bool) error {
-	if features.Enabled(cfg.FeatureGates, features.SelfHosting) && !IsControlPlaneSelfHosted(client) {
-
-		waiter := getWaiter(dryRun, client)
```

Comment:
This `getWaiter` is local here and we only call it in this spot, so better remove that too.

Reply:
:+1: